.github/workflows/main.yml

```diff
@@ -27,8 +27,8 @@ jobs:
           python-version: '3.6'
       - run: ./.github/scripts/install-drgn.sh
       - run: python3 -m pip install pylint pytest
-      - run: pylint -d duplicate-code sdb
```

Comment:
Why add `-d invalid name` here, vs. using a comment to disable this on the specific line where the error occurs? or better yet, changing the code so the error doesn't occur?

Reply:
Matt asked me about this when he first got the error and we talked about the various alternatives and it was me who suggested that he does that. I've hit this problem so many times when a variable like `i` or `p` would suffice (not only that but it would probably be a convention too - e.g. loops) but the linter complains.

I'm ok with changing the variables the majority of the time but sometimes you end up writing code like this:
```
for slab in slabs:
    ...
```
The only difference between the two identifiers above is `s` (plural vs singular). That is fine most of the time but it has happened to me where I copy pasted something and forgot to remove or add the `s`, and end up using `slab` outside of the loop or `slabs` within the loop without realizing it. That's just an example and eventually I can get used to it since it is very subjective, but I do find it annoying and less readable sometimes.

Reply:
Yea, I've run into that too.

I'm OK if we want to go with this approach. Leaving style decisions like this up to the tool, seems like a valid approach to me too; i.e. keep check in place. I'm good either way.

I think it would be nice to document why we're setting the `-d` options that we're setting; e.g. convert to using a `.pylintrc` file, and add the relevant commentary. I know I didn't do that originally with the `duplicate-code` setting, but capturing the motivation for our settings would be useful.

I'm OK opening a bug, and doing this in a follow on commit, if need be.

Reply:
I'll go ahead and resolve this issue, and open a bug to add some documentation for these settings.